Msfr updates (#18)
idaholab/virtual_test_bed: doc/content/msfr/griffin_pgh_results.md

# MSFR Turbulence Modeling Parameter Tuning Using Nek5000 Data

As a coarse mesh CFD code, Pronghorn requires high fidelity data to be able to

> model small scales phenomena. Turbulence modeling in Pronghorn currently supports

Reviewer: ```suggestion
model small scale phenomena. Turbulence modeling in Pronghorn currently supports
```

a parameter $\delta$, which is representative of the boundary layer thickness.
Using Nek5000 data, this parameter is tuned for Reynolds numbers 40K and 1M.
The latter is of particular interest to simulate the MSFR operational regime.

The mesh used for the tuning is the same as the one used in
[Nek5000 model](msfr/nek5000_cfd_model.md).

## Results for Reynolds number 40,000

The boundary conditions for this problem are velocity inlet profile, pressure
outlet, symmetry at the center of the geometry along the X axis and no slip
condition at the walls. Time derivatives are added to the momentum equation for
relaxation purposes.

Velocity magnitude and pressure fields obtained in Nek5000 and Pronghorn are
shown next.

!media media/msfr/Nek_Re40000_vfield.png
    style=width:30%
    caption=Nek5000 Velocity magnitude field for Re 40K.

!media media/msfr/PH_Re40000_vfield.png
    style=width:30%
    caption=Pronghorn Velocity magnitude field for Re 40K.

!media media/msfr/Nek_Re40000_pfield.png
    style=width:30%
    caption=Nek5000 Pressure field for Re 40K.

!media media/msfr/PH_Re40000_pfield.png
    style=width:30%
    caption=Pronghorn Pressure field for Re 40K.

> The parameter $\delta$ for this simulation is set to 0.12. This is the result

Reviewer: does it not have a unit

Author: yes, thank you

of a parametric study focused on minimizing the error of the velocity field,

> specifically at the minimum core radius.The velocity magnitude at the minimum

Reviewer: ```suggestion
specifically at the minimum core radius. The velocity magnitude at the minimum
```

## Results for Reynolds number 1,000,000

The boundary conditions for this problem are velocity inlet profile, pressure
outlet, symmetry at the center of the geometry along the X axis and standard
velocity wall functions at the walls. Time derivatives are added to the momentum
equation for relaxation purposes.

An exponentially decaying viscosity ramp is used for convergence purposes.

\begin{equation}
  \mu_{sim} = \mu_{phys} \cdot (exp(-3 \cdot t) + 1)
\end{equation}

Velocity magnitude and pressure fields obtained in Nek5000 and Pronghorn are
shown next.

!media media/msfr/Nek_Re1M_vfield.png
    style=width:30%
    caption=Nek5000 Velocity magnitude field for Re 1M.

!media media/msfr/PH_Re1M_vfield.png
    style=width:30%
    caption=Pronghorn Velocity magnitude field for Re 1M.

!media media/msfr/Nek_Re1M_pfield.png
    style=width:30%
    caption=Nek5000 Pressure field for Re 1M.

!media media/msfr/PH_Re1M_pfield.png
    style=width:30%
    caption=Pronghorn Pressure field for Re 1M.

The parameter $\delta$ for this simulation is set to 0.1. This is the result
of a parametric study focused on minimizing the error of the velocity field,

> specifically at the minimum core radius.The velocity magnitude at the minimum

Reviewer: ```suggestion
specifically at the minimum core radius. The velocity magnitude at the minimum
```

core radius is plotted in the following figure.

The behavior of the velocity fields are similar between both codes. The

> momentum diffusion at the core of the reactor is slightly higher in Pronghorn.

Reviewer: ```suggestion
momentum diffusion at the center of the reactor is slightly higher in Pronghorn.
```

!media media/msfr/Velocity_LinePlot_Re1M.png
    style=width:60%
    caption=Velocity magnitude at the minimum core radius.

> Pronghorn's total simulation time is 15 minutes using 4 cores.

Reviewer: say where time was spent. initialization?

Author: This is only navier stokes, so it begins from zero velocity field.

Author: It's the total simulation time

## General Remarks

> Even though Pronghorn presents the time edge it is important to note that in a

Reviewer: ```suggestion
Even though Pronghorn offers a lower computation time, it is important to note that in a
```

coarse mesh CFD code like Pronghorn, the tuning done in this section is
dependent on the mesh due to the numerical diffusion. The importance of this
numerical diffusion could be reduced in the future with the implementation of
second order upwind discretization (currently using linear upwind) and the
use of non-orthogonal correctors for meshes with a degree of skewness.

# MSFR Griffin-Pronghorn Results

## Initialization

The high Reynolds number in the MSFR is challenging for convergence purposes.
Thus, before running the full multiphysics model, converged velocity fields for
the operational regime are obtained and outputted to an exodus file

> ("msfr_SS_wallfunctions.e"). The $\delta$ value is obtained from the comparison
> against Nek5000.

Reviewer: ```suggestion
("msfr_SS_wallfunctions.e").
```
this is odd here

Reviewer: make a small paragraph about it maybe. What else did you import from the pure fluids simulation?

Reviewer: and why did you do (because you don't have coupled Nek-Griffin)

Author: Switched the initialization position here. It's a little clearer. Let me know what you think

The steady state velocity magnitude fields for the MSFR Griffin-Pronghorn model
is depicted below.

!media media/msfr/PH_Griffin_SteadyState_NS.png
       style=width:30%

## Full Model Steady State

The Griffin-Pronghorn model predicts the following steady-state velocity and
temperature fields in the MSFR:

!media media/msfr/PH_Griffin_v_SS.png
       style=width:30%

!media media/msfr/PH_Griffin_T_SS.png
       style=width:30%

The maximum temperature of the simulation is around 1280 K. Note that despite  
some momentum diffusion occurring in the reactor core, the flow at the center is

>  low where the power density has a maximum. This leads to high temperatures in
>  the center region, which in turn drives the flow vertically through buoyancy.
>  We can also observe flow separation along the shield wall. The salt here will
>  be in direct contact with the core structure and may challenge the temperature
>  limitations of the structural material.

Reviewer: ```suggestion
low where the power density has a maximum. This leads to high temperatures in
the center region, which in turn drives the flow vertically through buoyancy.
We can also observe flow separation along the shield wall. The salt here will
be in direct contact with the core structure and may challenge the temperature
limitations of the structural material.
```

Reviewer: Cut this paragraph in two. Need more info about the vertical flow, and the missing horizontal flow

Also flow separation or re-circulation? Flow separation leading to same effect, low V ?

Author: It could be expressed as either. In general when the boundary layer is dettached from the wall in an expansion we can call it flow separation. Then there might be a certain.degree of recirculation in the low flow area.
